blogposts/inside-rust/01-announcement.md

```diff
@@ -0,0 +1,162 @@
+    * However, because unwinding is now part of most system ABIs, even C
+      functions can unwind -- most notably, `pthread_cancel` can cause all
+      manner of C functions to unwind, including common functions like `read`
+      and `write`.  (For a complete list, search for "cancellation points" in
+      the [pthreads man
+      page](http://man7.org/linux/man-pages/man7/pthreads.7.html).)
+* Changing the behavior from `-Cpanic=unwind` to `-Cpanic=abort` should not
+  cause Undefined Behavior.
+  * However, this may not be tenable, or at least not without making binaries
+    much larger. See the discussion below for more details.
+  * It may, of course, cause programs to abort that used to execute
+    successfully. This could occur if a panic would've been caught and
+    recovered.
+
+## The primary question: should the `"C"` ABI permit unwinding?
+
+The core question that we would like to decide is whether the `"C"` ABI, as
+defined by Rust, should permit unwinding.
+
+This is not a question we expected to be debating. We've long declared that
+unwinding through Rust's `"C" ABI is undefined behavior. In part, this is
+because nobody had spent the time to figure out what the correct behavior would
+be, or how to implement it, although (as we'll see shortly) there are other
+good reasons for this choice. 
+
+In any case, in PR #65646, @Amanieu proposed that we could, in fact, simply
+define the behavior of unwinding across `"C"` boundaries. In discussing this,
+discovered that the question of whether the `"C" ABI should permit unwinding was
+less clear-cut than we had assumed.
+
+If the `"C"` ABI does not permit unwinding, a new ABI, called `"C unwind"`,
+will be introduced specifically to support unwinding.
+
+## Three specific proposals
+
+The project group has narrowed the design space down to three specific
+proposals. Two of these introduce the new `"C unwind"` API, and one does not.
+
+Each proposal specifies the behavior of each type of unwind (Rust `panic!`,
+foreign (e.g. C++), and forced (e.g. `pthread` cancel)) when it encounters an
+ABI boundary under either the `panic=unwind` or `panic=abort` compile-mode.
+
+### 2 APIs, minimal specification
+
+* `"C"` ABI boundary, `panic=<any>`
+  * `panic!`: program aborts
+  * forced unwind, no destructors: unwind behaves normally
+  * all other foreign unwinding: undefined behavior
+* `"C unwind"` ABI boundary
+  * With `panic=unwind`: all types of unwinding behave normally
+  * With `panic=abort`: all types of unwinding abort the program
+
+### 2 APIs, forced unwinding always permitted
+
+This is the same as the previous design, except that when compiled with
+`panic=abort`, forced exceptions would *not* be intercepted; that is, the
+program would not abort.
+
+### 1 API
+
+* `panic=unwind`: unwind behaves normally
+* `panic=abort`:
+  * `panic!`: program aborts
+  * forced unwind, no destructors: unwind behaves normally
+  * all other foreign unwinding: undefined behavior
```

Comment:
My understanding was that any unwinding, whether foreign or forced, is acceptable as long as there are no destructors. It would be helpful in the table to separate "foreign unwinding, destructors" and "foreign unwinding, no destructors".

Reply:
@Amanieu That wasn't my understanding of the Dropbox Paper doc, but I found it somewhat hard to follow so long after the meeting.

`panic=abort` will not generate landing pads on `"C"` ABI functions. As long as there aren't destructors in those frames, is that sound?

Reply:
The idea with option 3 is that we don't make any distinction between force unwind and other foreign exceptions: they are all treated the same way.

Reply:
I think the idea was that we'd behave like `-fno-exceptions`, basically, and hence I think I agree with @Amanieu